Add edit and delete actions in InitiativeType admin table

andreslucena opened the pull request:
#### :tophat: What? Why?

In admin's Initiative Type  we don't show the Edit and Delete icons as we already do in other sections. This PR adds them.

While working on this, I also found that the Initiative Type form didn't show an error when saving the form without adding the Banner image. This is also fixed here. 

#### :pushpin: Related Issues

- Fixes  #9150

#### Testing

##### Initiative type icons

1. Sign in as admin 
2. Create an initiative type
3. Go to the table and see that you have the actions Edit (Pencil icon) and also Delete (with the X icon) 

##### Initiative type banner image button

1. Sign in as admin 
2. Go to the /admin/initiatives_types and click on "New initiative type" button 
3. Fill the title and description
4. Click on Create button
5. See that there is an error in the banner image (before there wasn't) 

### :camera: Screenshots

![image](https://user-images.githubusercontent.com/717367/162441610-fc9b652b-4e37-4344-af13-9d713af7d1c3.png)

:hearts: Thank you!

andreslucena commented:
I had the same question, actually 😄. For sure, it feels weird making a fix and not adding a spec. My main doubt was that I'm not sure if we're testing the action icons in these tables, at least couldn't find any other specs for this. 

Mind that this is only about adding the icons/actions, we already have tests for the controllers of course:
https://github.com/decidim/decidim/blob/746e49ae71754059f6c684c563d99ba46d8eed5e/decidim-initiatives/spec/controllers/decidim/initiatives/admin/initiatives_type_scopes_controller_spec.rb#L111
https://github.com/decidim/decidim/blob/746e49ae71754059f6c684c563d99ba46d8eed5e/decidim-initiatives/spec/controllers/decidim/initiatives/admin/initiatives_type_scopes_controller_spec.rb#L193

ahukkanen commented:
> My main doubt was that I'm not sure if we're testing the action icons in these tables, at least couldn't find any other specs for this.

Yes we are testing them in other places.

E.g. for proposals:
https://github.com/decidim/decidim/blob/746e49ae71754059f6c684c563d99ba46d8eed5e/decidim-proposals/spec/system/admin/admin_edits_proposal_spec.rb#L28-L31

https://github.com/decidim/decidim/blob/746e49ae71754059f6c684c563d99ba46d8eed5e/decidim-proposals/spec/system/admin/admin_edits_proposal_spec.rb#L51-L55

Maybe we could add similar tests to `decidim-initiatives/spec/system/admin/admin_manages_initiatives_spec.rb`?

andreslucena commented:
Oh, I didn't know about them. Awesome, I'll add them

andreslucena commented:
I think I found a better spec for implementing this, https://github.com/decidim/decidim/blob/a4fa48245c32be5b5a2ab48eccb6f95cdc404406/decidim-initiatives/spec/system/admin/initiative_types_controller_spec.rb. I couldn't find it before because the naming wasn't consistent with the rest of specs. 

In the process, I've made a few more changes: 
1. I've renamed it, so it was consistent with the others specs
2. I've cleaned it up a bit 
3. Until now, it was using the URLs directly, now I think it makes a bit more sense using the actions/icons 
4. I've also renamed initiatives_type_scopes_controller_spec.rb to be consistent also

ahukkanen commented:
Let me know when this is ready for re-review @andreslucena !

andreslucena commented:
> Let me know when this is ready for re-review @andreslucena !

Yes, it's ready to be re-reviewed @ahukkanen
